stdlib/ballerina-http/src/main/ballerina/http/authz_handler.bal

```diff
@@ -58,26 +59,40 @@ function HttpAuthzHandler::handle (string username, string serviceName, string r
             // match against those.
             string[] authCtxtScopes = runtime:getInvocationContext().userPrincipal.scopes;
             if (lengthof authCtxtScopes > 0) {
-                boolean authorized = matchScopes(scopes, authCtxtScopes);
-                if (authorized) {
-                    log:printDebug("Successfully authorized to access resource: " + resourceName + ", method: " +
-                            method);
-                } else {
-                    log:printDebug("Authorization failure for resource: " + resourceName + ", method: " + method);
-                }
+                boolean authorized = checkForScopeMatch(scopes, authCtxtScopes, resourceName, method);
                 // cache authz result
                 self.cacheAuthzResult(authzCacheKey, authorized);
                 return authorized;
             } else {
-                // no scopes found for user, authorization failure
-                log:printDebug("No scopes found for user: " + username + " to access resource: " + resourceName + ",
+                // no scopes found for user, try to retrieve using the auth provider
+                string[] scopesFromAuthProvider = self.authProvider.getScopes(username);
+                if (lengthof scopesFromAuthProvider > 0) {
+                    boolean authorized = checkForScopeMatch(scopes, scopesFromAuthProvider, resourceName, method);
+                    // cache authz result
+                    self.cacheAuthzResult(authzCacheKey, authorized);
+                    return authorized;
+                } else {
+                    log:printDebug("No scopes found for user: " + username + " to access resource: " + resourceName + ",
                                                                                                     method:" + method);
```

Comment:
Can we change the formatting here so that the `"method:"` and rest comes to this line?

Reply:
If we do that, we will be going past the column length limit of 120.

Reply:
What I meant was -

```
log:printDebug("No scopes found for user: " + username + " to access resource: " + resourceName + 
    ", method:" + method);
```